arlo_server/models.py

```diff
@@ -332,3 +332,16 @@ class File(db.Model):
     name = db.Column(db.String(250), nullable=False)
     contents = db.Column(db.Text, nullable=False)
     uploaded_at = db.Column(db.DateTime(timezone=False), nullable=False)
+
+    # Metadata for processing files in the background.
+    processing_started_at = db.Column(db.DateTime(timezone=False), nullable=True)
+    processing_succeeded_at = db.Column(db.DateTime(timezone=False), nullable=True)
```

Comment:
suggest `processing_completed_at` with `processing_error` null if successful, not-null if not. Otherwise data model could be inconsistent, with both `succeeded_at` and `errored_at` being non-null.

Reply:
This crossed my mind but didn't think through it all the way. I like it, will do.